projects/plugins/jetpack/modules/scan/class-admin-sidebar-link.php

```diff
@@ -155,7 +155,7 @@ private function should_show_backup() {
 	private function has_scan() {
 		$this->maybe_refresh_transient_cache();
 		$scan_state = get_transient( 'jetpack_scan_state' );
-		return ! $scan_state || 'unavailable' !== $scan_state->state;
+		return $scan_state && 'unavailable' !== $scan_state->state;
```

Comment:
Could there be a point where the transient hasn't been set yet, even though a scan plan was purchased? As far as I can tell, the transient is only set when you visit the Jetpack dashboard in wp-admin, and thus call the API endpoint that ends up setting the transient. Same thing with `jetpack_rewind_state`. 

Is it possible for someone to purchase a plan and then go back to wp-admin without visiting the Jetpack dashboard? That would be my worry with this change.

Reply:
The current logic allows links to show up whenever the transient doesn't exist or expires which causes performance issues (we are loading things that we shouldn't) and bad UX as these links show up (and disappear) every time transient expires and gets regenerated.

>As far as I can tell, the transient is only set when you visit the Jetpack dashboard in wp-admin, and thus call the API endpoint that ends up setting the transient. Same thing with jetpack_rewind_state.

If the transient doesn't exist a cron job is set to create it and that works for any wp-admin page.  

https://github.com/Automattic/jetpack/blob/af0dda38e06dcc9810aed03d6b582bae8448c091/projects/plugins/jetpack/modules/scan/class-admin-sidebar-link.php#L194

That should allow for the transient to get created on any visit to wp-admin and allow links to show up in the next step. 

I'm not sure if there is a reason this was done via cron job. Could we just trigger `Jetpack_Core_Json_Api_Endpoints::get_scan_state();` when a missing transient is detected?

Also, it looks like the transient gets updated with a new state (in case someone purchases a plan) only when visiting the Jetpack dashboard pages which doesn't sound like an ideal solution. 

That being said, with the scan transient, things get even more complicated as it seems that the fallback code in class-admin-bar-notice.php is also acting as a trigger for transient creation. I'm just not sure how exactly.
I was looking into this as a part of the related issue #22668

https://github.com/Automattic/jetpack/blob/8bd3f270b57486316920d4eaf927d4062aec3492/projects/plugins/jetpack/modules/scan/class-admin-bar-notice.php#L94

Reply:
Just to be sure when you say "the transient is only set when you visit the Jetpack dashboard in wp-admin, and thus call the API endpoint that ends up setting the transient" you mean the code in class-scan.php is doing the API call?